### ISSUE TITLE
testing tflite quantized model on an input image

### ISSUE THREAD

**Issue by sumeetssaurav** [tensorflow/tensorflow#23113]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: No
- TensorFlow installed from (source or binary): Binary
- TensorFlow version (use command below): tf_nightly
- Python version:  3.5.2
- Bazel version (if compiling from source): NA
- GCC/Compiler version (if compiling from source):  5.4.0
- CUDA/cuDNN version: 9.0/7.1
- GPU model and memory: GeFORCE 1080 Ti and 12 GB

You can collect some of this information using our environment capture [script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with
python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

**Describe the current behavior**

**Describe the expected behavior**

**Code to reproduce the issue**
 NA

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

`import numpy as np
import tensorflow as tf
import pathlib
import cv2
from keras.preprocessing import image
from keras.applications.resnet50 import preprocess_input, decode_predictions
img_path = 'elephant.jpg'
img = image.load_img(img_path, target_size=(299, 299))

x = image.img_to_array(img)
input_data = np.expand_dims(x, axis=0)

#print(input_data.type)
#x = preprocess_input(x)

archive_path = tf.keras.utils.get_file("resnet_v2_101.tgz", "https://storage.googleapis.com/download.tensorflow.org/models/tflite_11_05_08/resnet_v2_101.tgz", extract=True)
archive_path = pathlib.Path(archive_path)
archive_dir = str(archive_path.parent)
print(archive_dir)

graph_def_file = pathlib.Path(archive_path).parent/"resnet_v2_101_299_frozen.pb"
input_arrays = ["input"] 
output_arrays = ["output"]

converter = tf.contrib.lite.TFLiteConverter.from_frozen_graph(
  str(graph_def_file), input_arrays, output_arrays, input_shapes={"input":[1,299,299,3]})

converter.post_training_quantize = False
resnet_tflite_file = graph_def_file.parent/"resnet_v2_101.tflite"
resnet_tflite_file.write_bytes(converter.convert())

converter.post_training_quantize = True
resnet_tflite_file = graph_def_file.parent/"resnet_v2_101_quantized.tflite"
resnet_tflite_file.write_bytes(converter.convert())

#!ls -lh {archive_dir}/*.tflite

# Load TFLite model and allocate tensors.
interpreter = tf.contrib.lite.Interpreter(model_path="/home/sumeet/.keras/datasets/resnet_v2_101_quantized.tflite")
interpreter.allocate_tensors()

# Get input and output tensors.
input_index= interpreter.get_input_details()
print(input_index)
#output_details = interpreter.get_output_details()
output_index = interpreter.get_output_details()
print(output_index)

# Test model on random input data.
input_shape = input_index[0]['shape']
#input_data = np.array(np.random.random_sample(input_shape), dtype=np.float32)
#input_data = np.array(X(input_shape), dtype=np.float32)
interpreter.set_tensor(input_index[0]['index'],input_data)

interpreter.invoke()
predictions = interpreter.get_tensor(output_index[0]['index'])
predictions[0]['name']
print(predictions[0,0])
#print('Predicted:', decode_predictions(predictions, top=3)[0])`

How to make predictions on a custom input image like the code given below
from keras.applications.resnet50 import ResNet50
from keras.preprocessing import image
from keras.applications.resnet50 import preprocess_input, decode_predictions
import numpy as np

model = ResNet50(weights='imagenet')

img_path = 'elephant.jpg'
img = image.load_img(img_path, target_size=(224, 224))
x = image.img_to_array(img)
x = np.expand_dims(x, axis=0)
x = preprocess_input(x)

preds = model.predict(x)
print(preds.shape)
# decode the results into a list of tuples (class, description, probability)
# (one such list for each sample in the batch)
print('Predicted:', decode_predictions(preds, top=3)[0])
# Predicted: [(u'n02504013', u'Indian_elephant', 0.82658225), (u'n01871265', u'tusker', 0.1122357), (u'n02504458', u'African_elephant', 0.061040461)

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Exact command to reproduce

**Comment by ymodak** [comment]
Can you please provide following information? Thanks
Describe the current behavior.
Describe the expected behavior.

**Comment by tofulawrence** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by caiya55** [comment]
Hi, if you are looking for some object detection and can be converted to .tflite file. I think this is great.https://coral.withgoogle.com/models/. google released this model for its new hardware device.  But I am still trying to convert my own model. I'd like to know after converting to tflite model, how to keep your tflite output still float32 type.
